sklego/meta.py

```diff
@@ -474,3 +476,65 @@ def predict_proba(self, X):
 
     def score(self, X, y):
         return self.estimator_.score(X, y)
+
+
+class ConfusionBalancer(BaseEstimator, MetaEstimatorMixin):
+    """
+    The ConfusionBalancer attempts to give it's child estimator a more balanced
+    output by learning from the confusion matrix during training. The idea is that
+    the confusion matrix calculates P(C_i | M_i) where C_i is the actual class and
+    M_i is the class that the underlying model gives. We use these probabilities to
+    attempt a more balanced prediction by averaging the correction from the confusion
+    matrix with the original probabilities.
+
+    .. math::
+        p(\text{class_j}) = \alpha p(\text{model}_j) + (1-\alpha) p(\text{class_j} | \text{model}_j) p(\text{model}_j)
+
+    :param model: a scikit learn compatible classification model that has predict_proba
+    :param alpha: a hyperparameter between 0 and 1, determines how much to apply smoothing
+    """
+    def __init__(self, model, alpha: float = 0.5):
+        self.model = model
+        self.alpha = alpha
+
+    def _is_classifier(self):
+        return any(['ClassifierMixin' in p.__name__ for p in type(self.model).__bases__])
+
+    def fit(self, X, y):
+        """
+        Fit the data.
+
+        :param X: array-like, shape=(n_columns, n_samples,) training data.
+        :param y: array-like, shape=(n_samples,) training data.
+        :return: Returns an instance of self.
+        """
+        X, y = check_X_y(X, y, estimator=self.model, dtype=FLOAT_DTYPES)
+        if not self._is_classifier():
```

Comment:
We have a `ProbabilisticClassifier` in `base` which does the check you need here so you can use `isinstance(self.model, base.ProbabilisticClassifier)` i think

Reply:
yeah i noticed that later on, should be fixed now.